### PR TITLE
Eslint update

### PR DESCRIPTION
**Pull request by svillafe**
## Why are you doing this?

Updated `eslint-plugin-jsx-a11y` and `eslint-config-airbnb` libs. Applying the new eslint to the files.

Related PR: https://github.com/guardian/support-frontend/pull/292

### CODE REVIEW

File: assets/components/contribAmounts/contribAmounts.jsx [hunk 1]
```diff
@@ -280,7 +282,7 @@ function getAttrs(props: PropTypes): ContribAttrs {
 
   if (props.contribType === 'ANNUAL') {
 
-    const userDefined = props.contribAmount.annual.userDefined;
+    const { userDefined } = props.contribAmount.annual;
```
Comment (Ap0c) [inline assets/components/contribAmounts/contribAmounts.jsx]: It really recommends we do this? This looks less clear to me...

File: assets/components/radioToggle/radioToggle.jsx [hunk 2]
```diff
@@ -40,6 +40,7 @@ export default function RadioToggle(props: PropTypes) {
     const className = getClassName(props);
     const accessibilityHintId = `accessibility-hint-${radioId}`;
     const accessibilityHint = <p id={accessibilityHintId} className="accessibility-hint">{radio.accessibilityHint}</p>;
+    /* eslint-disable jsx-a11y/label-has-for */
```
Comment (Ap0c) [inline assets/components/radioToggle/radioToggle.jsx]: Why is this disabled? `htmlFor` is present so there should be no lint error surely?

File: assets/pages/regular-contributions/helpers/ajax.js [hunk 3]
```diff
@@ -72,8 +74,7 @@ function requestData(paymentFieldName: PaymentField,
   }
 
   const ophanIds: OphanIds = getOphanIds();
-  const referrerAcquisitionData: ReferrerAcquisitionData =
-    getState().common.referrerAcquisitionData;
+  const { referrerAcquisitionData } = getState().common;
```
Comment (Ap0c) [inline assets/pages/regular-contributions/helpers/ajax.js]: We've lost a type here no?

File: assets/helpers/logger.js [hunk 4]
```diff
@@ -5,7 +5,7 @@ import Raven from 'raven-js';
 
 export const init = () => {
   const dsn: string = 'https://dc13eb8698614a8081ce6a139d9f4aab@sentry.io/171710';
-  const gitCommitId = window.guardian.gitCommitId;
+  const { gitCommitId } = window.guardian;
```
Comment (JustinPinner) [inline assets/helpers/logger.js]: Wait, what? Is `window.guardian` only the `gitCommitId`?

File: assets/helpers/tracking/acquisitions.js [hunk 5]
```diff
@@ -67,15 +67,14 @@ export type Campaign = $Keys<typeof campaigns>;
 // Retrieves the user's campaign, if known, from the campaign code.
 function getCampaign(acquisition: ReferrerAcquisitionData): ?Campaign {
 
-  const campaignCode = acquisition.campaignCode;
+  const { campaignCode } = acquisition;
```
Comment (JustinPinner) [inline assets/helpers/tracking/acquisitions.js]: Ah, now I see what this is doing. I don't like it though!

File: assets/helpers/url.js [hunk 6]
```diff
@@ -63,7 +62,7 @@ const addQueryParamToURL = (urlOrPath: string, paramsKey: string, paramsValue: ?
 // Retrieves the domain for the given env, e.g. guardian.com/gulocal.com.
 function getBaseDomain(): Domain {
 
-  const origin = window.location.origin;
+  const { origin } = window.location;
```
Comment (JustinPinner) [inline assets/helpers/url.js]: Angry face

File: assets/helpers/logger.js (hunk 4 above)
Comment (Ap0c) [reply assets/helpers/logger.js]: The `gitCommitId` is in the `window.guardian` object.

File: assets/components/contribAmounts/contribAmounts.jsx (hunk 1 above)
Comment (svillafe) [reply assets/components/contribAmounts/contribAmounts.jsx]: This is enforced by `eslint-config-airbnb`.

File: assets/components/radioToggle/radioToggle.jsx (hunk 2 above)
Comment (svillafe) [reply assets/components/radioToggle/radioToggle.jsx]: That rule also enforces that `label` should wrap the `input` and this was broken. Therefore, I disabled for these lines and I created a card to fix it in the future since it was getting a huge PR.

**Comment by svillafe** [comment]
Selenium tests are passing in DEV.

File: assets/helpers/logger.js (hunk 4 above)
Comment (svillafe) [reply assets/helpers/logger.js]: The new set of rules enforce to use [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) whenever is possible. If we as a team don't like it we can turn it off in the future. Since this PR is quite big, I would like to merge with this property on, discuss the best approach and then turn it off or keep it on in a follow up PR.
